crates/basilica-validator/src/rental/mod.rs

```diff
@@ -100,37 +53,45 @@ fn parse_ssh_host(credentials: &str) -> Result<&str> {
     Ok(host)
 }
 impl RentalManager {
-    /// Create a new rental manager
-    pub fn new(miner_client: Arc<MinerClient>, persistence: Arc<SimplePersistence>) -> Self {
+    /// Helper function to create a ContainerClient with SSH credentials
+    fn create_container_client(&self, ssh_credentials: &str) -> Result<ContainerClient> {
+        let private_key_path = self
+            .ssh_key_manager
+            .as_ref()
+            .and_then(|km| km.get_persistent_key())
+            .map(|(_, path)| path.clone());
+
+        ContainerClient::new(ssh_credentials.to_string(), private_key_path)
+    }
+
+    /// Create a new rental manager with SSH key manager
+    pub fn new(
+        miner_client: Arc<MinerClient>,
+        persistence: Arc<SimplePersistence>,
+        ssh_key_manager: Arc<ValidatorSshKeyManager>,
+    ) -> Self {
         let deployment_manager = Arc::new(DeploymentManager::new());
         let log_streamer = Arc::new(LogStreamer::new());
-        let (tx, rx) = mpsc::channel(100);
-        let health_monitor = Arc::new(HealthMonitor::new(tx));
 
-        // Start the health status monitor
-        HealthStatusMonitor::start_monitoring(rx, persistence.clone());
+        // Create health monitor with SSH key manager
+        let health_monitor = Arc::new(DatabaseHealthMonitor::new(
+            persistence.clone(),
+            ssh_key_manager.clone(),
+        ));
+
+        // Start the monitoring loop
+        health_monitor.clone().start_monitoring_loop();
```

Comment:
i think this is a bad pattern, to start the service with `::new()` it's not expected, it should be with the `start()`.

I agree, with the not `start_rental()` but should have it's own explicit `start_monitor` method.